view/ViewableData.php

```diff
@@ -249,6 +249,12 @@ public function castingHelperPair($field) {
 	 */
 	public function castingHelper($field) {
 		if($this->hasMethod('db') && $fieldSpec = $this->db($field)) {
+			Deprecation::notice(
+				'4.0',
+				'ViewableData::castingHelper() will no longer extract casting information "db". Please override
+				castingHelper in your ViewableData subclass.',
+				Deprecation::SCOPE_GLOBAL
+			);
 			return $fieldSpec;
```

Comment:
I figured this “db” functionality is better placed in `DataObject::castingHelper()` now that I’ve added that method, so it can be removed from here. In theory someone might be relying on it for a `ViewableData` subclass, so I’ve taken the safer route of deprecating it instead of removing it now.